src/librustc_trans/trans/common.rs

```diff
@@ -656,6 +657,14 @@ impl<'blk, 'tcx> BlockS<'blk, 'tcx> {
                                          self.fcx.param_substs,
                                          value)
     }
+
+    pub fn build(&self) -> Builder<'blk, 'tcx> {
+        // Create a fresh builder from the crate context.
+        let builder = self.ccx().builder();
+        // Set the builder's position to this block's end.
+        builder.position_at_end(self.llbb);
```

Comment:
Each crate has only one builder, which `BuilderRef` points to. I'm worried the lack of repositioning could introduce bugs. For example, after the call to `MirContext.unreachable_block`, which may emit an instruction, the builder for bcx must be repositioned (I think).

Also, keeping builder refs alongside blocks in the table would add at least several bytes of overhead for each block. Otherwise, you'd need a different dance every time bcx is passed from MIR to common trans.